node/bft/src/gateway.rs

```diff
@@ -443,59 +425,11 @@ impl<N: Network> Gateway<N> {
         }
     }
 
-    /// Returns the maximum number of connected peers.
-    pub fn max_connected_peers(&self) -> usize {
-        self.tcp.config().max_connections as usize
-    }
-
     /// Returns the list of connected addresses.
     pub fn connected_addresses(&self) -> HashSet<Address<N>> {
         self.get_connected_peers().into_iter().map(|peer| peer.aleo_addr).collect()
     }
 
-    /// Attempts to connect to the given peer IP.
-    pub fn connect(&self, peer_ip: SocketAddr) -> Option<JoinHandle<bool>> {
-        // Return early if the attempt is against the protocol rules.
-        if let Err(forbidden_error) = self.check_connection_attempt(peer_ip) {
-            warn!("{forbidden_error}");
-            return None;
-        }
-
-        let self_ = self.clone();
-        Some(tokio::spawn(async move {
-            debug!("Connecting to validator {peer_ip}...");
-            // Attempt to connect to the peer.
-            match self_.tcp.connect(peer_ip).await {
-                Ok(_) => true,
-                Err(error) => {
-                    warn!("Unable to connect to '{peer_ip}' - {error}");
-                    false
-                }
-            }
-        }))
-    }
-
-    /// Ensure we are allowed to connect to the given peer.
-    fn check_connection_attempt(&self, peer_ip: SocketAddr) -> Result<()> {
-        // Ensure the peer IP is not this node.
-        if self.is_local_ip(peer_ip) {
-            bail!("{CONTEXT} Dropping connection attempt to '{peer_ip}' (attempted to self-connect)")
```

Comment:
We're losing this CONTEXT information, right? I think it is fine, but not sure if some people rely on it. 

If it is needed, we could add something like  `PeerPoolHandling::pool_type` or `PeerPoolHandling::pool_name` to the trait.

Reply:
fixed in https://github.com/ProvableHQ/snarkOS/pull/3894/commits/85cd265eae216593ea2ea6b77a75749910cf5988